src/frontend/packages/store/src/effects/entity-fetcher.effect.ts

```diff
@@ -0,0 +1,231 @@
+  return pagination.pageRequests[pagination.currentPage] || {
+    busy: false,
+    error: false,
+    message: ''
+  };
+}
+
+export const TRY_ENTITY_FETCH = '[Entity Service] Try Entity Fetch';
+export class TryEntityFetchAction implements Action {
+  constructor(
+    public fetchFn: (key?: string) => void,
+    public entityKey: string,
+    public entityId: string,
+  ) { }
+  public type = TRY_ENTITY_FETCH;
+}
+
+export const TRY_ENTITY_VALIDATION = '[Entity Service] Try Entity Validation';
+export class TryEntityValidationAction implements Action {
+  constructor(
+    public entityKey: string,
+    public entityId: string,
+    public cFAction: ICFAction
+  ) { }
+  public type = TRY_ENTITY_VALIDATION;
+}
+
+export const TRY_PAGINATION_FETCH = '[Pagination Service] Try Pagination Fetch';
+export class TryEntityPaginationFetchAction implements Action {
+  constructor(
+    public paginationKey: string,
+    public entityKey: string,
+    public isLocal: boolean,
+    public actions: PaginatedAction[]
+  ) { }
+  public type = TRY_PAGINATION_FETCH;
+}
+
+export const TRY_PAGINATION_VALIDATION = '[Pagination Service] Try Pagination Validation';
+export class TryEntityPaginationValidationAction implements Action {
+  constructor(
+    public paginationKey: string,
+    public entityKey: string,
+    public isLocal: boolean,
+    public actions: PaginatedAction[]
+  ) { }
+  public type = TRY_PAGINATION_VALIDATION;
+}
+
+@Injectable()
+export class TryFetchEffect {
+
+  constructor(
+    private actions$: Actions,
+    private store: Store<AppState>
+  ) { }
+
+  @Effect({ dispatch: false })
+  tryFetch$ = this.actions$.pipe(
+    ofType<TryEntityFetchAction>(TRY_ENTITY_FETCH),
+    mergeMap((action: TryEntityFetchAction) => {
+      return combineLatest(
+        this.store.select(selectRequestInfo(action.entityKey, action.entityId)).pipe(first()),
+        this.store.select(selectEntity(action.entityKey, action.entityId)).pipe(first(), startWith(null)),
+        observableOf(action)
+      );
+    }),
+    tap(([requestInfo, entity, action]) => {
+      if (this.shouldCallAction(requestInfo, entity)) {
+        action.fetchFn();
+      }
+    })
+  );
+
+  @Effect({ dispatch: false })
+  tryValidation$ = this.actions$.pipe(
+    ofType<TryEntityValidationAction>(TRY_ENTITY_VALIDATION),
+    mergeMap((action) => {
+      return this.store.select(selectRequestInfo(action.entityKey, action.entityId)).pipe(
+        filter(entityRequestInfo => !isEntityBlocked(entityRequestInfo)),
+        first(),
+        tap(() => {
+          this.store.dispatch(new ValidateEntitiesStart(
+            action.cFAction,
+            [action.entityId],
+            false
+          ));
+        })
+      );
+    })
+  );
+
+  @Effect({ dispatch: false })
+  tryPaginationFetch$ = this.actions$.pipe(
+    ofType<TryEntityPaginationValidationAction>(TRY_PAGINATION_VALIDATION),
```

Comment:
Should be `ofType<TryEntityPaginationFetchAction>(TRY_PAGINATION_FETCH),`. Tested with/without and doesn't actually make any difference due both actions being called at the same time with similar params